apps/video/test/marionette/play_multiple_types_test.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+
+var assert = require('assert'),
+    Video = require('./lib/video');
+
+marionette('play various types of videos', function() {
+
+  var app, client, selectors;
+
+  client = marionette.client({
+    profile: {
+      prefs: {
+        'device.storage.enabled': true,
+        'device.storage.testing': true,
+        'device.storage.prompt.testing': true
+      }
+    }
+  });
+
+  setup(function() {
+    // Remove all files in temp device storage.
+    client.fileManager.removeAllFiles();
+    selectors = Video.Selector;
+    app = new Video(client);
+  });
+
+  function verifyVideo(type, done) {
+    var title = 'VID_0001';
+    var filename = title + '.' + type;
+
+    client.fileManager.add({
+      type: 'videos',
+      filePath: 'test_media/Movies/' + filename
+    });
+    app.launch();
+    app.tapThumbnail(0);
```

Comment:
I don't actually understand why this works. The launch() method does not wait for metadata parsing to complete, so when launch() returns there might not be any thumbnails displayed.  tapThumbnail() does a findElements() to find all thumbnails.Does that method wait until at least one match is found? I suppose it probably does. But notice that this code would not work reliably if you tried tapThumbnail(1). Because findElements() might return with an array that only had the first thumbnail.

Anyway, my point here is that it might be nice to have an explicit test here to see if there are any thumbnails. If metadata parsing fails, this line will just timeout after 20 seconds, I think. You'll get a test failure but not an informative error message.   If we're not able to play the video then we're probably not able to create a thumbnail for it.  If you try running the test with a corrupt or unsupported video file, it will probably fail here with a timeout.  

So it would be nice if there was some other way to wait for metadata parsing to be complete and then check to see how many thumbnails were created without blocking.

If you don't want to fix all this, how about just simplifying and commenting:

```
// Wait for a thumbnail to appear and then tap it.
// Note that if this test fails it will probably fail with a timeout on this call
app.thumbnail.tap();
```